File: reverse_ binary_tree/reverse_binary_tree.go
package main

import "fmt"

type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

func main(){
	root := &TreeNode{
		Val: 1,
	}

	node2 := &TreeNode{
		Val: 2,
	}

	node3 := &TreeNode{
		Val: 3,
	}

	node4 := &TreeNode{
		Val: 4,
	}

	node5 := &TreeNode{
		Val: 5,
	}

	node6 := &TreeNode{
		Val: 6,
	}

	node3.Left = node2
	node3.Right = node4
	root.Left = node3
	root.Right = node5
	node5.Right = node6

	reverseBinaryTree(root)
	fmt.Println(root)
}

func reverseBinaryTree(node *TreeNode) *TreeNode {
	if node == nil {
		return nil
	}
	aux := node.Left;
	node.Left = reverseBinaryTree(node.Right)
	node.Right = reverseBinaryTree(aux)
	return node
}

File: reverse_ binary_tree/reverse_binary_tree.ts
class TreeNode {
  public left: TreeNode | null = null;
  public right: TreeNode | null = null;

  constructor(public value: number) {}
}

const root = new TreeNode(1);
const node2 = new TreeNode(2);
const node3 = new TreeNode(3);
const node4 = new TreeNode(4);
const node5 = new TreeNode(5);
const node6 = new TreeNode(6);
const node7 = new TreeNode(7);

root.left = node3;
root.right = node6;

node3.left = node2;
node3.right = node4;

node6.left = node5;
node6.right = node7;

const reversex = (node: TreeNode | null) => {
  if (node === null) return node;

  const auxLeft = node.left;

  node.left = reversex(node.right);
  node.right = reversex(auxLeft);
  return node;
};

console.log(reversex(root));
